### PR TITLE
Support platform group ID parametrization

### PR DESCRIPTION
**Pull request by pjgg**
Currently the annotation `@GitRepositoryQuarkusApplication` supports the following injections in mavenArgs annotation property:

* -Dquarkus.platform.version=${QUARKUS_VERSION} 
* -Dquarkus-plugin.version=${QUARKUS-PLUGIN_VERSION} 

Note: if `quarkus-plugin.version` and/or `quarkus.platform.version` is not provided then a default version is set

This PR add a new mavenArg injection:
* -Dquarkus.platform.group-id=${QUARKUS_PLATFORM_GROUP-ID}

If `quarkus.platform.group-id` is not provided then `io.quarkus` would be used as a default group

Usage example:

```
@DisabledOnNative(reason = "This scenario is using uber-jar, so it's incompatible with Native")
@QuarkusScenario
public class TodoDemoIT {
    @GitRepositoryQuarkusApplication(repo = "https://github.com/quarkusio/todo-demo-app.git", mavenArgs = "-Dquarkus.package.type=uber-jar -DskipTests=true -Dquarkus.platform.group-id=${QUARKUS_PLATFORM_GROUP-ID} -Dquarkus.platform.version=${QUARKUS_VERSION} -Dquarkus-plugin.version=${QUARKUS-PLUGIN_VERSION}")
    static final RestService app = new RestService();

    @Test
    public void verify() {
        app.given()
                .get()
                .then()
                .statusCode(HttpStatus.SC_OK);
    }
}
```

Prompt

```
mvn -U clean verify -Dall-modules -Dopenshift -pl external-applications -Dmaven.repo.local=~/Downloads/rh-quarkus-2.2.5.CR3-maven-repository/maven-repository -Dquarkus.platform.group-id=com.redhat.quarkus.platform -Dquarkus.platform.artifact-id=quarkus-bom -Dquarkus.platform.version=2.2.5.Final-redhat-00007 -Dquarkus-plugin.version=2.2.5.Final-redhat-00007 
```

This example will use the external application `https://github.com/quarkusio/todo-demo-app.git` with the following maven args:
* quarkus.platform.version=${QUARKUS_VERSION} -> taken from promt -> 2.2.5.Final-redhat-00007
* quarkus-plugin.version=${QUARKUS-PLUGIN_VERSION} -> taken from promt -> 2.2.5.Final-redhat-00007
* quarkus.platform.group-id=${QUARKUS_PLATFORM_GROUP-ID} -> taken from promt -> com.redhat.quarkus.platform

### CODE REVIEW

**Comment by mjurc** [comment]
The new test is failing in all PR builds - @pjgg would you kindly review?

**Comment by pjgg** [comment]
run tests

**Comment by pjgg** [comment]
run tests

File: examples/external-applications/src/test/java/io/quarkus/qe/QuickstartUsingDefaultsIT.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 @DisabledOnOs(value = OS.WINDOWS, disabledReason = "Windows does not support long file paths")
 public class QuickstartUsingDefaultsIT {
 
-    @GitRepositoryQuarkusApplication(repo = "https://github.com/quarkusio/quarkus-quickstarts.git", contextDir = "getting-started")
+    @GitRepositoryQuarkusApplication(repo = "https://github.com/quarkusio/quarkus-quickstarts.git", contextDir = "getting-started", branch = "development")
```
Comment (mjurc) [inline examples/external-applications/src/test/java/io/quarkus/qe/QuickstartUsingDefaultsIT.java:16]: I am not sure I understand purpose behind using `development` branch.

File: examples/external-applications/src/test/java/io/quarkus/qe/QuickstartUsingUsingUberJarIT.java [hunk 2]
```diff
@@ -15,11 +15,11 @@
 @DisabledOnOs(value = OS.WINDOWS, disabledReason = "Windows does not support long file paths")
 public class QuickstartUsingUsingUberJarIT {
 
-    @GitRepositoryQuarkusApplication(repo = "https://github.com/quarkusio/quarkus-quickstarts.git", contextDir = "getting-started", mavenArgs = "-Dquarkus.package.type=uber-jar")
+    @GitRepositoryQuarkusApplication(repo = "https://github.com/quarkusio/quarkus-quickstarts.git", contextDir = "getting-started", mavenArgs = "-Dquarkus.package.type=uber-jar -DskipTests=true -Dquarkus.platform.group-id=${QUARKUS_PLATFORM_GROUP-ID} -Dquarkus.platform.version=${QUARKUS_VERSION} -Dquarkus-plugin.version=${QUARKUS-PLUGIN_VERSION}", branch = "development")
```
Comment (mjurc) [inline examples/external-applications/src/test/java/io/quarkus/qe/QuickstartUsingUsingUberJarIT.java:18]: Same here - why `development` branch? It is not stable.

**Review by mjurc — CHANGES_REQUESTED** [review]
Besides the `development` branch change, this looks good to me.

**Review by mjurc — APPROVED** [review]

**Pull request merged**